createdisk.sh

```diff
@@ -274,7 +300,12 @@ ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- sudo podman tag $certImage open
 ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- 'sudo crictl images -q | xargs -n 1 sudo crictl rmi 2>/dev/null'
 
 # Replace pull secret with a null json string '{}'
-${OC} --kubeconfig $1/auth/kubeconfig replace -f pull-secret.yaml
+# If its an OKD install we don't need to remove as that's not a real pull-secret
+if [ ${OKD_INSTALL} == false]; then
+    ${OC} --kubeconfig $1/auth/kubeconfig replace -f pull-secret.yaml
+    # Remove pull secret from the VM
+    ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- 'sudo rm -f /var/lib/kubelet/config.json'
+fi
```

Comment:
This can be ignored in case of FCOS because there is no pull secret but need to check if there is configmap for same. Remove pull secret from the VM can be ignored because it just delete the file if exist if not it will just run the `rm` command and do nothing.

Reply:
Both the file and config map exists in FCOS too, we just don't need to remove them, as those are dummy, you still need a valid json in the install-config for the pull-secret.